### PR TITLE
first commit

### PR DESCRIPTION
**Pull request by RuslanIbIsh**

### CODE REVIEW

**Review by Ilya-brinchuk — APPROVED** [review]

File: src/main/java/com/iri/movietickets/model/validator/PasswordsEqualConstraintValidator.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class PasswordsEqualConstraintValidator implements
+        ConstraintValidator<PasswordsEqualConstraint, Object> {
+
+    @Override
+    public boolean isValid(Object user, ConstraintValidatorContext arg1) {
+        UserRequestDto userRequestDto = (UserRequestDto) user;
+        return userRequestDto.getPassword().equals(userRequestDto.getRepeatPassword());
```
Comment (KseniiaMakarova) [inline src/main/java/com/iri/movietickets/model/validator/PasswordsEqualConstraintValidator.java:13]: what if password is null?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/iri/movietickets/controller/MovieSessionController.java [hunk 2]
```diff
@@ -48,7 +49,7 @@ public void create(@RequestBody MovieSessionRequestDto movieSessionRequestDto) {
 
     @PutMapping("/{id}")
     public void update(@PathVariable Long id,
-                       @RequestBody MovieSessionRequestDto movieSessionRequestDto) {
+                       @RequestBody @Valid MovieSessionRequestDto movieSessionRequestDto) {
```
[the remark this answers is not in the document]
Comment (RuslanIbIsh) [reply src/main/java/com/iri/movietickets/controller/MovieSessionController.java:52]: done

File: src/main/java/com/iri/movietickets/model/validator/CustomEmailValidator.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class CustomEmailValidator
+        implements ConstraintValidator<CustomEmailConstraint, String> {
+
+    @Override
+    public boolean isValid(String emailField,
+                           ConstraintValidatorContext cxt) {
+        return emailField != null && emailField.matches(".+@.+\\..+")
```
[the remark this answers is not in the document]
Comment (RuslanIbIsh) [reply src/main/java/com/iri/movietickets/model/validator/CustomEmailValidator.java:12]: done

File: src/main/java/com/iri/movietickets/controller/CustomGlobalExceptionHandler.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@ControllerAdvice
+public class CustomGlobalExceptionHandler extends ResponseEntityExceptionHandler {
+
```
[the remark this answers is not in the document]
Comment (RuslanIbIsh) [reply src/main/java/com/iri/movietickets/controller/CustomGlobalExceptionHandler.java:18]: done

File: src/main/java/com/iri/movietickets/model/validator/PasswordsEqualConstraintValidator.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+import com.iri.movietickets.model.dto.UserRequestDto;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class PasswordsEqualConstraintValidator implements
+        ConstraintValidator<PasswordsEqualConstraint, Object> {
+
+    @Override
+    public boolean isValid(Object user, ConstraintValidatorContext arg1) {
```
[the remark this answers is not in the document]
Comment (RuslanIbIsh) [reply src/main/java/com/iri/movietickets/model/validator/PasswordsEqualConstraintValidator.java:11]: done

File: src/main/java/com/iri/movietickets/model/validator/CustomEmailValidator.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class CustomEmailValidator
+        implements ConstraintValidator<CustomEmailConstraint, String> {
+
+    @Override
+    public boolean isValid(String emailField,
+                           ConstraintValidatorContext cxt) {
```
[the remark this answers is not in the document]
Comment (RuslanIbIsh) [reply src/main/java/com/iri/movietickets/model/validator/CustomEmailValidator.java:11]: done

File: src/main/java/com/iri/movietickets/model/validator/PasswordsEqualConstraintValidator.java (hunk 1 above)
Comment (RuslanIbIsh) [reply src/main/java/com/iri/movietickets/model/validator/PasswordsEqualConstraintValidator.java:13]: done

File: src/main/java/com/iri/movietickets/model/validator/PasswordsEqualConstraintValidator.java [hunk 7]
```diff
@@ -0,0 +1,15 @@
+package com.iri.movietickets.model.validator;
+
+import com.iri.movietickets.model.dto.UserRequestDto;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class PasswordsEqualConstraintValidator implements
+        ConstraintValidator<PasswordsEqualConstraint, Object> {
+
```
[the remark this answers is not in the document]
Comment (RuslanIbIsh) [reply src/main/java/com/iri/movietickets/model/validator/PasswordsEqualConstraintValidator.java:9]: done

**Review by KseniiaMakarova — APPROVED** [review]

File: src/main/java/com/iri/movietickets/model/validator/CustomEmailValidator.java [hunk 8]
```diff
@@ -0,0 +1,15 @@
+package com.iri.movietickets.model.validator;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class CustomEmailValidator
+        implements ConstraintValidator<CustomEmailConstraint, String> {
+
```
[the remark this answers is not in the document]
Comment (RuslanIbIsh) [reply src/main/java/com/iri/movietickets/model/validator/CustomEmailValidator.java:9]: done

**Pull request merged**